pandas/core/computation/parsing.py

```diff
@@ -6,11 +6,14 @@
 from keyword import iskeyword
 import token
 import tokenize
-from typing import Iterator, Tuple
+from typing import TYPE_CHECKING, Iterator, Tuple
 
 # A token value Python's tokenizer probably will never use.
 BACKTICK_QUOTED_STRING = 100
 
+if TYPE_CHECKING:
+    from pandas._typing import Label
```

Comment:
should be able to import this unconditionally

Reply:
Agreed, but I thought one should still import conditionally if the imports are only needed for typing. It can save on import time, no? Happy to remove the conditional if that's what we prefer.

Reply:
I've removed the conditional import.

Reply:
> It can save on import time, no?

In principle, but you really only get a noticeable bump if you avoid importing a given module _anywhere_.  so once we've imported from pandas._typing in one module, we might as well do it everywhere else.  (i did a lot of import time optimization about a year ago)

> It can also avoid import cycles in the future

we've structured it so that pandas._typing is strictly "before" everything in core (reflected in our isort config)